tag-system/src/main/java/org/togetherjava/tjbot/tag/TagSystemInstanceHolder.java

```diff
@@ -0,0 +1,33 @@
+package org.togetherjava.tjbot.tag;
+
+import org.togetherjava.tjbot.db.Database;
+
+public class TagSystemInstanceHolder {
+    private TagSystemInstanceHolder() {}
+
+    private static TagSystem instance;
+
+    public static TagSystem getOrCreate(Database database) {
```

Comment:
I am not sure why you need all these methods, where do you use them?

Reply:
To create a lazy-singleton that lets you specify the database instance.

Reply:
This isnt safe btw. May be just use an enum instead
Visibility of instance isn't guaranteed without it being volatile.

Reply:
Even now it isn't safe. Check for double checked locking if you want to do it this way.
I would still prefer something like an enum or may a static class holding the instance if you want to do it lazily